src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs

```diff
@@ -1703,9 +1703,14 @@ internal static unsafe char ParseFormatSpecifier(ReadOnlySpan<char> format, out
                     // digits.  Further, for compat, we need to stop when we hit a null char.
                     int n = 0;
                     int i = 1;
-                    while (i < format.Length && (((uint)format[i] - '0') < 10) && n < 10)
+                    while (i < format.Length && (((uint)format[i] - '0') < 10))
                     {
-                        n = (n * 10) + format[i++] - '0';
+                        int temp = ((n * 10) + format[i++] - '0');
+                        if (temp < n)
+                        {
+                            ThrowHelper.ThrowFormatException_BadFormatSpecifier();
```

Comment:
Good discussion here. FWIW, I had the regular `throw` statement here first because I didn't see any `ThrowHelpers` defined already. I changed it back to the canonical `throw` statement now. 

P.S: The auto-generation work sounds awesome. I'm not a fan of the `ThrowHelper` pattern. It feels like a place where we try to generate better code than the compiler.